From d05ff19ed3f3bb6bc914135531e4bdb6d332adce Mon Sep 17 00:00:00 2001
From: Jiping Ma <jiping.ma2@windriver.com>
Date: Thu, 11 Apr 2024 02:48:47 +0000
Subject: [PATCH] mlnx-ofa_kernel: fix bnxt_re build error

NETDEV_CHANGEINFODATA already exists as enum 25 in kernel header file
netdevice.h, but it is defined 0x0019 here, so that NETDEV_CHANGEINFODATA
and NETDEV_BONDING_INFO is same value that caused the following errors.

bnxt_re-227.0.130.0/main.c: In function bnxt_re_task:
bnxt_re-227.0.130.0/main.c:5668:2: error: duplicate case value
 5668 |  case NETDEV_CHANGEINFODATA:
      |  ^~~~
bnxt_re-227.0.130.0/main.c:5642:2: note: previously used here
 5642 |  case NETDEV_BONDING_INFO:
      |  ^~~~
bnxt_re-227.0.130.0/main.c: In function bnxt_re_netdev_event:
bnxt_re-227.0.130.0/main.c:5908:2: error: duplicate case value
 5908 |  case NETDEV_CHANGEINFODATA:
      |  ^~~~
bnxt_re-227.0.130.0/main.c:5870:2: note: previously used here
 5870 |  case NETDEV_BONDING_INFO:
      |  ^~~~

Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 include/linux/netdevice.h | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/include/linux/netdevice.h b/include/linux/netdevice.h
index 277e5a0..756809d 100644
--- a/include/linux/netdevice.h
+++ b/include/linux/netdevice.h
@@ -24,10 +24,6 @@
     ( (netdev)->ethtool_ops = (ops) )
 #endif
 
-#ifndef NETDEV_BONDING_INFO
-#define NETDEV_BONDING_INFO     0x0019
-#endif
-
 static inline int netdev_set_master(struct net_device *dev,
 				    struct net_device *master)
 {
-- 
2.43.0

